sky/shell/platform/android/platform_view_android.cc

```diff
@@ -343,10 +344,16 @@ class AndroidGLContext {
   FTL_DISALLOW_COPY_AND_ASSIGN(AndroidGLContext);
 };
 
-static jlong Attach(JNIEnv* env, jclass clazz, jint skyEngineHandle) {
+static jlong Attach(JNIEnv* env,
+                    jclass clazz,
+                    jint skyEngineHandle,
+                    jobject flutterView) {
   PlatformViewAndroid* view = new PlatformViewAndroid();
   view->ConnectToEngine(mojo::InterfaceRequest<SkyEngine>(
       mojo::ScopedMessagePipeHandle(mojo::MessagePipeHandle(skyEngineHandle))));
+  // Remember the JNIEnv and create a weak reference to the flutterView Java
```

Comment:
"Remember the JNIEnv" is probably obsolete?  RunFromSource will attach the VM to get a JNIEnv